Errors when building tests relying on stdexec

## Description
Errors appeared when I was trying to build tests about `std::execution`.
```
In file included from /Users/Nan/GSoC/hpx/libs/core/execution_base/include/hpx/execution_base/sender.hpp:13,
                 from /Users/Nan/GSoC/hpx/libs/core/execution_base/include/hpx/execution_base/this_thread.hpp:13,
                 from /Users/Nan/GSoC/hpx/libs/core/execution_base/src/agent_ref.cpp:9:
/Users/Nan/GSoC/hpx/libs/core/execution_base/include/hpx/execution_base/stdexec_forward.hpp:165:35: error: 'dependent_domain' has not been declared in 'stdexec'
  165 |     HPX_CXX_EXPORT using stdexec::dependent_domain;
      |                                   ^~~~~~~~~~~~~~~~
/Users/Nan/GSoC/hpx/libs/core/execution_base/include/hpx/execution_base/stdexec_forward.hpp:240:35: error: 'make_completion_signatures' has not been declared in 'stdexec'
  240 |     HPX_CXX_EXPORT using stdexec::make_completion_signatures;
      |                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/Nan/GSoC/hpx/libs/core/execution_base/include/hpx/execution_base/stdexec_forward.hpp:245:35: error: 'transform_env' has not been declared in 'stdexec'
  245 |     HPX_CXX_EXPORT using stdexec::transform_env;
      |                                   ^~~~~~~~~~~~~
In file included from /Users/Nan/GSoC/hpx/libs/core/execution_base/include/hpx/execution_base/any_sender.hpp:12,
                 from /Users/Nan/GSoC/hpx/build/libs/core/execution_base/include/hpx/modules/execution_base.hpp:17,
                 from /Users/Nan/GSoC/hpx/libs/core/execution_base/include/hpx/execution_base/this_thread.hpp:87:
/Users/Nan/GSoC/hpx/libs/core/execution_base/include/hpx/execution_base/completion_signatures.hpp:512:54: warning: 'using stdexec::empty_env = struct stdexec::__env::env<>' is deprecated: stdexec::empty_env is now spelled stdexec::env<> [-Wdeprecated-declarations]
  512 |         typename Env = hpx::execution::experimental::empty_env>
      |                                                      ^~~~~~~~~
In file included from /Users/Nan/GSoC/hpx/build/_deps/stdexec-src/include/stdexec/execution.hpp:18,
                 from /Users/Nan/GSoC/hpx/libs/core/execution_base/include/hpx/execution_base/stdexec_forward.hpp:35:
/Users/Nan/GSoC/hpx/build/_deps/stdexec-src/include/stdexec/__detail/__execution_fwd.hpp:102:9: note: declared here
  102 |   using empty_env [[deprecated("stdexec::empty_env is now spelled stdexec::env<>")]] = env<>;
      |         ^~~~~~~~~
/Users/Nan/GSoC/hpx/libs/core/execution_base/include/hpx/execution_base/completion_signatures.hpp:520:54: warning: 'using stdexec::empty_env = struct stdexec::__env::env<>' is deprecated: stdexec::empty_env is now spelled stdexec::env<> [-Wdeprecated-declarations]
  520 |         typename Env = hpx::execution::experimental::empty_env>
      |                                                      ^~~~~~~~~
/Users/Nan/GSoC/hpx/build/_deps/stdexec-src/include/stdexec/__detail/__execution_fwd.hpp:102:9: note: declared here
  102 |   using empty_env [[deprecated("stdexec::empty_env is now spelled stdexec::env<>")]] = env<>;
      |         ^~~~~~~~~
/Users/Nan/GSoC/hpx/libs/core/execution_base/include/hpx/execution_base/completion_signatures.hpp:537:54: warning: 'using stdexec::empty_env = struct stdexec::__env::env<>' is deprecated: stdexec::empty_env is now spelled stdexec::env<> [-Wdeprecated-declarations]
  537 |         typename Env = hpx::execution::experimental::empty_env>
      |                                                      ^~~~~~~~~
/Users/Nan/GSoC/hpx/build/_deps/stdexec-src/include/stdexec/__detail/__execution_fwd.hpp:102:9: note: declared here
  102 |   using empty_env [[deprecated("stdexec::empty_env is now spelled stdexec::env<>")]] = env<>;
      |         ^~~~~~~~~
gmake[3]: *** [libs/core/execution_base/CMakeFiles/hpx_execution_base.dir/build.make:79: libs/core/execution_base/CMakeFiles/hpx_execution_base.dir/src/agent_ref.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/Makefile2:111243: libs/core/execution_base/CMakeFiles/hpx_execution_base.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:54907: libs/core/algorithms/tests/unit/algorithms/CMakeFiles/uninitialized_relocate_sender_test.dir/rule] Error 2
gmake: *** [Makefile:6068: uninitialized_relocate_sender_test] Error 2
```

I suspect the issue stems from recent changes in [NVIDIA's upstream stdexec implementation](https://github.com/NVIDIA/stdexec).
We need to make some adjustments to ensure compatibility.

## Steps to Reproduce
1. Build HPX
```
cmake -S . -B build \
  -DCMAKE_INSTALL_PREFIX="$HOME/hpx_install" \
  -DCMAKE_BUILD_TYPE=Debug \
  -DHPX_WITH_EXAMPLES=OFF \
  -DHPX_WITH_TESTS=ON \
  -DHPX_WITH_CXX_STANDARD=20 \
  -DHPX_WITH_STDEXEC=ON \
  -DHPX_WITH_FETCH_ASIO=ON \
  -DCMAKE_C_COMPILER="$(brew --prefix)/bin/gcc-13" \
  -DCMAKE_CXX_COMPILER="$(brew --prefix)/bin/g++-13"
```

2. Build test
```
cmake --build build --target uninitialized_relocate_sender_test > build.log 2>&1
```

## Environment
OS Platform and Distribution: macOS 15.6.1

Build System: CMake 4.2.0

Hardware Architecture: Apple M4

--- a/libs/core/execution_base/include/hpx/execution_base/stdexec_forward.hpp
+++ b/libs/core/execution_base/include/hpx/execution_base/stdexec_forward.hpp
@@ -162,7 +162,6 @@ namespace hpx::execution::experimental {
 
     // Domain
     HPX_CXX_EXPORT using stdexec::default_domain;
-    HPX_CXX_EXPORT using stdexec::dependent_domain;
 
     // Execute
     HPX_CXX_EXPORT using stdexec::execute;
@@ -237,12 +236,10 @@ namespace hpx::execution::experimental {
     HPX_CXX_EXPORT using stdexec::sends_stopped;
     HPX_CXX_EXPORT using stdexec::value_types_of_t;
 
-    HPX_CXX_EXPORT using stdexec::make_completion_signatures;
     HPX_CXX_EXPORT using stdexec::transform_completion_signatures;
     HPX_CXX_EXPORT using stdexec::transform_completion_signatures_of;
 
     // Transform sender
-    HPX_CXX_EXPORT using stdexec::transform_env;
     HPX_CXX_EXPORT using stdexec::transform_sender;
     HPX_CXX_EXPORT using stdexec::transform_sender_result_t;
     HPX_CXX_EXPORT using stdexec::transform_sender_t;

--- a/libs/core/executors/include/hpx/executors/thread_pool_scheduler.hpp
+++ b/libs/core/executors/include/hpx/executors/thread_pool_scheduler.hpp
@@ -298,6 +298,20 @@ namespace hpx::execution::experimental {
                 {
                     return e.sched;
                 }
+
+                // clang-format off
+                template <typename CPO,
+                    HPX_CONCEPT_REQUIRES_(
+                        meta::value<meta::one_of<
+                            CPO, set_value_t, set_stopped_t>>
+                    )>
+                // clang-format on
+                constexpr auto query(
+                    hpx::execution::experimental::get_completion_scheduler_t<
+                        CPO>) const noexcept
+                {
+                    return sched;
+                }
             };
 
             friend constexpr env tag_invoke(
@@ -353,6 +367,11 @@ namespace hpx::execution::experimental {
             thread_pool_policy_scheduler const& sched)
         {
             return {sched};
+        }
+
+        constexpr sender<thread_pool_policy_scheduler> schedule() const
+        {
+            return {*this};
         }
 
         void policy(Policy policy) noexcept

